src/libs/nav.js

```diff
@@ -51,13 +51,14 @@ export const Redirector = ({ pathname, search }) => {
   return null
 }
 
-const parseRoute = (handlers, { pathname, search }) => {
+const parseRoute = (handlers, { pathname, search, state }) => {
   const handler = _.find(({ regex }) => regex.test(pathname), handlers)
   console.assert(handler, 'No handler found for path')
   return handler && {
     ...handler,
     params: _.zipObject(handler.keys, _.tail(handler.regex.exec(pathname))),
-    query: qs.parse(search, { ignoreQueryPrefix: true, plainObjects: true })
+    query: qs.parse(search, { ignoreQueryPrefix: true, plainObjects: true }),
+    state: state || {}
```

Comment:
I like the idea of adding a new param and leaving the others untouched. This may allow us to have a cleaner url, and still have enough data to display the relevant information.

> I don't think importing the same snapshots would be a bookmarked feature, it seems like kind of a one time feature. Do you agree?

For the most part, yes. I was thinking of a case where another user wants the same data, imported to different workspaces over time. But it is unlikely they would know to bookmark the URL for this purpose.- and it is more of an edge case.

> Can users share workspaces? If so, that might be more useful than sharing the import page
Yes - that is the typical flow.

Either way, if we load the title data from localstorage or via a fetch when it is not available it should work with a bookmark.

Reply:
> Either way, if we load the title data from localstorage or via a fetch when it is not available it should work with a bookmark.
If we stored it in localstorage, would we have an issue here where the title and header show our header/title ("catalog, link datasets") if the user got to the same page from a different link?

Re: Loading in data and storing it in localstorage
Would this solution use the `useDataCatalog` method? if the values are NOT there? And if it does, we're going to have to add logic in the application to explicitly pull out the field "dct:title". Is this okay? Would we be tying this solution down too closely to our structure of a snapshot?

Reply:
What do you think about the current solution btw? Using state? Do you think it's bad?

Reply:
I think your instinct to avoid breaking refresh is a good one, so I'd lean toward putting the IDs in the URL. If you are worried about the implementation being coupled to the data catalog, just make that coupling explicit; for example, you could call the parameter `dcSnapshotIds`.